src/test/java/jenkins/branch/harness/MultiBranchImpl.java

```diff
@@ -55,11 +57,18 @@ protected MultiBranchImpl(ItemGroup parent, String name) {
         return new BranchProjectFactoryImpl();
     }
 
+    @Override
+    public boolean scheduleBuild() {
+        LOGGER.info("Indexing multibranch project: " + getDisplayName());
+        return super.scheduleBuild();
+    }
+
     public static class BranchProjectFactoryImpl extends BranchProjectFactory<FreeStyleProject, FreeStyleBuild> {
 
         @Override
         public FreeStyleProject newInstance(Branch branch) {
             FreeStyleProject job = new FreeStyleProject(getOwner(), branch.getName());
+            job.onCreatedFromScratch();
```

Comment:
Not sure, something you need to debug. The Javadoc for `newInstance` explicit forbids you from calling `onCreatedFromScratch`, since this is called by `MultiBranchProject`.